src/components/TextField/TextField.jsx

```diff
@@ -22,15 +32,21 @@ const TextField = ({ id, label, placeholder, type, onChange, value, isDisabled,
           rows={5}
         />
       ) : (
-        <input
-          className="textfield-input"
-          id={id}
-          placeholder={placeholder}
-          type={type}
-          value={value}
-          onChange={onChange}
-          disabled={isDisabled}
-        />
+        <div className="searchInput">
+          <input
+            className="textfield-input"
+            id={id}
+            placeholder={placeholder}
+            type={type}
+            value={value}
+            onChange={onChange}
+            disabled={isDisabled}></input>
+          {value && (
+            <button className="searchClear" onClick={onClear}>
+              X
+            </button>
+          )}
```

Comment:
```suggestion
          {value && onClear && (
            <button className="searchClear" onClick={onClear}>
              X
            </button>
          )}
```

this way we are sure that `onClear` is passed and our code wont break, because we dont want every button to be cleared.